### DISCUSSION TITLE
How to identify coupling coefficients in a coupled system of equations

### DISCUSSION THREAD

**Discussion by wanikhawar** [dynamicslab/pysindy#301] [Q&A] [answered]
I have a system of coupled van der pol oscillators.

$\ddot{x_1}+\lambda_1 (1-x_1^2) \dot{x_1} + \alpha_1 x_1 = \beta_1 (x_1-x_2)$
$\ddot{x_2}+\lambda_2 (1-x_2^2) \dot{x_2} + \alpha_2 x_2 = \beta_2 (x_2-x_1)$

I want to get the values of the coefficients but whenever I use pySINDy, the coefficients of $x_1$ and $x_2$ are added up as the algorithm has no way of knowing the system has this particular form. How do I get the values of $\beta_1$ and $\beta_2$ without them mixing up with $\alpha_1$ and $\alpha_2$

**Comment by akaptano** [comment, marked as answer]
There is no need to disambiguate: just write your equations above as (beta1 -alpha1) x1 - beta1 x2 and similar for the second equation. Then they have the form you want. The only way to disambiguate alpha1 and beta1 is using the fact that the coefficient of x2 in the first equation is beta1, so you can infer alpha1 from the coefficient of the x1 term.

**Reply by wanikhawar** [reply]
Thanks a lot.
